src/H5Tpkg.h

```diff
@@ -150,9 +150,17 @@ struct H5T_stats_t {
     H5_timevals_t times;  /*total time for conversion	     */
 };
 
+/* Context struct for information used during datatype conversions */
+typedef struct H5T_conv_ctx_t {
```

Comment:
New structure that holds relevant information during conversion to make conversions on compound and other container types faster. Similar to `H5T_cdata_t`, but that structure is public since it gets passed to an app conversion callback and the information we want here is partly private.